src/cljx/schema/spec/variant.cljx

```diff
@@ -13,7 +13,17 @@
   (let [g (:guard o)
         c (spec/sub-checker o params)
         step (if g
-               (fn [x] (if (g x) (c x) (else x)))
+               (fn [x]
+                 (macros/try-catchall
+                  (if (g x)
```

Comment:
is there merit to tightening the try-catch to be just around `(g x)` and not around sub-checker/else?